Weekly_Work/Week1/Code_changes-git/Metrics_Root_class.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import json
+import datetime
+import matplotlib.pyplot as plt
+import pandas as pd
+
+
+# In[7]:
+
+
+class Metric():
+    
+    def __init__(self, path, since=None, until=None, source_code_exclude_list=None):
+
+        self.since = since
+        self.until = until
+        self.source_code_exclude_list = source_code_exclude_list
+        self.repo_urls = set()
+        
+        clean_commit_list = list()
+        clean_issue_list = list()
+        clean_pr_list = list()
+        
+        with open(path, 'r') as raw_data:
+            for line in raw_data:
+                line = json.loads(line)
+                self.repo_urls.add(line['origin'])
+                clean_line = dict()
+                if line['category'] == "commit":
+                    clean_line = self._clean_commit(line)
+
+                    if self._is_source_code(clean_line):
+                        clean_commit_list.append(clean_line)
+                    
+
+                elif line['category'] == "issue":
+                    if "pull_request" not in line['data']:
+                        clean_line = self._clean_issue(line)
+                        clean_issue_list.append(clean_line)
+                    else: continue
+                    
+
+                elif line['category'] == "pull_request":
+                    clean_line = self._clean_pr(line)
+                    clean_pr_list.append(clean_line)
+                        
+                        
+                self.clean_commit_df = pd.DataFrame(clean_commit_list)
+                self.clean_issue_df = pd.DataFrame(clean_issue_list)
+                self.clean_pr_df = pd.DataFrame(clean_pr_list)
+                
+            self.clean_dict = {
+                'commit': self.clean_commit_df,
+                'issue': self.clean_issue_df,
+                'pull_request': self.clean_pr_df
+            }
+       
+        if self.since:
```

Comment:
It would be better to have a period of time (maybe as a tuple (since, until) as a parameter to the metrics, where one of the values could be None, meaning "since always" or "forever". But this is a minor detail: the main rationale is that it could be more easily mapped to the "period" parameter that we have in the description of the metrics.

Reply:
Good idea!